app/controllers/video-list.controller.js

```diff
@@ -0,0 +1,69 @@
+(function() {
+  'use strict';
+
+  class VideoListCtrl {
+    constructor(StorageService, VideoService) {
+      Object.assign(this, {StorageService, VideoService});
+      this.StorageService.get('videos').then(videos => {
+        this.videos = JSON.parse(JSON.stringify(videos)) || [];
+      });
+      this.StorageService.get('videos').then(videos => {

```

Comment:
po co robisz dwa razy requesta do StorageService?
generalnie wysylanie requestow o zewnetrzne dane to najwieksza bolaczka frontend developmentu wiekszych aplikacji, bo nagle musisz puscic 20-30 requestow do serwera i:
a. mulisz serwer,
b. Twoj Klient czeka wiecznosc na odpowiedz.

jak pobralas te dane w 7 linii to tam je przeprocesuj :)

Reply:
dodatkowo: nie ogarniam tej akcji z temporaryVideos. po co to wlasciwie jest?